pkg/k8s/controller.go

```diff
@@ -216,25 +243,50 @@ func (c *Controller) getObjectEntities(obj interface{}, selector port.Selector,
 		return nil, fmt.Errorf("error converting from unstructured: %v", err)
 	}
 
-	var selectorResult = true
-	if selector.Query != "" {
-		selectorResult, err = jq.ParseBool(selector.Query, structuredObj)
-		if err != nil {
-			return nil, fmt.Errorf("invalid selector query '%s': %v", selector.Query, err)
+	entities := make([]port.Entity, 0, len(mappings))
+	if itemsToParse != "" {
```

Comment:
this if is redundant
instead always runs the loop and set the single value to an array in case there is no items to parse

Reply:
I need the if so I can check wether the items array parse is needed, but I need fix it to use only one loop as you suggest with the nice idea of insert the single object into array.
Let me know if that what you ment.

Reply:
Apparently I had a problem with this solution, that because the items were overriding each other so only the last item was mapped to entity in the later for loop.
I revert it back to map the item to an entity in each item so it would not happened.
Another solution will be to clone each item to the edited object before insert it to the objectsToMap array but there is no build in clone method for Go so I figured it is a bit overkill.

Reply:
we talked about this

Reply:
Done